### PR TITLE
Update to 0.10.3

### PR DESCRIPTION
**Pull request by bouweandela**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by bouweandela** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -17,15 +17,31 @@ build:
...
+    - filelock >=3.0
+    - mypy >=0.50
+    - pytest >=4.6
+    - python >=3.6
 
 test:
+  source_files:
+    - pyproject.toml
+    - tests
+  requires:
```
Comment (xhochy) [inline recipe/meta.yaml:35]:
```suggestion
  requires:
    - pip
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -17,15 +17,31 @@ build:
...
+    - mypy
+    - pytest >=4.6
+    - packaging
+    - pexpect
+    - pytest-cov
+    - pytest-randomly
+    - pytest-xdist
+  commands:
```
Comment (xhochy) [inline recipe/meta.yaml:43]:
```suggestion
  commands:
    - pip check
```

**Review by xhochy — COMMENTED** [review]
Let's modernize it completely.

**Review by xhochy — APPROVED** [review]

**Pull request merged**